lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2149,39 +2149,31 @@ export class UserAgentApplication {
      * Utils function to create the Authentication
      * @param {@link account} account object
      * @param scopes
-     * @param silentCall
      *
-     * @returns {string} token type: id_token or access_token
+     * @returns {string} token type: token, id_token or id_token token
      *
      */
-    private getTokenType(accountObject: Account, scopes: string[], silentCall: boolean): string {
-        /*
-         * if account is passed and matches the account object/or set to getAccount() from cache
-         * if client-id is passed as scope, get id_token else token/id_token_token (in case no session exists)
-         */
-        let tokenType: string;
+    private getTokenType(accountObject: Account, scopes: string[]): string {
+        // Check if account in request matches account in cache
+        const accountsMatch = Account.compareAccounts(accountObject, this.getAccount());
+        // Check if login scopes are included
+        const containsLoginScopes = ScopeSet.isLoginScopes(scopes, this.clientId);
 
-        // acquireTokenSilent
-        if (silentCall) {
-            if (Account.compareAccounts(accountObject, this.getAccount())) {
-                tokenType = (scopes.indexOf(this.config.auth.clientId) > -1) ? ResponseTypes.id_token : ResponseTypes.token;
-            }
-            else {
-                tokenType  = (scopes.indexOf(this.config.auth.clientId) > -1) ? ResponseTypes.id_token : ResponseTypes.id_token_token;
-            }
-            return tokenType;
+        // response_type is id_token because only login scopes are present (["openid", "profile"].length == 2);
+        if (containsLoginScopes && (scopes.length == 2)) {
```

Comment:
I would rather explicitly check that the scopes array only contains `openid` and `profile` to avoid a buggy scenario where scopes list is ["User.Read", "openid"]

Reply:
Given the way #1681 validates scopes, there is no way this code is reached with either login scope without appending the other. Adding that explicit check feels like repeating validation, which I believe should be a responsibility of the API rather than this method. The bug in question would only occur if the validation behavior changes. Having said this, if you still think having the explicit check is necessary, I have no other reason to disagree.

Reply:
Since this is a utils function I think it's a good idea to not assume anything has been done before it. If you think the logic is duplicated I would suggest writing a helper function you can call in both places.

Reply:
Added a new static method to ScopeSet called `containsOIDCScopes` that does a hard check for both `openid` and `profile`, replaced the call to `isLoginScopes` here with a call to `containsOIDCScopes`. Now the possible bug you mentioned should be mitigated. Thanks!